metricbeat/module/elasticsearch/index/data.go

```diff
@@ -249,6 +252,23 @@ func addClusterStateFields(idx *Index, clusterState mapstr.M) error {
 		return fmt.Errorf("failed to get index routing table from cluster state: %w", err)
 	}
 
+	indexMetadata, err := getClusterStateMetricForIndex(clusterState, idx.Index, "metadata")
+	if err != nil {
+		return fmt.Errorf("failed to get index metadata from cluster state: %w", err)
+	}
+
+	indexTierPreference, err := getIndexTierPreferenceFromMetadata(indexMetadata)
+	if err != nil {
```

Comment:
Not all indices have tier preference in their settings, why would that be an error? Shouldn't we just ignore those indices?